### DISCUSSION TITLE
How to perform some initialization operations when the main window opens?

### DISCUSSION THREAD

**Discussion by alkene5** [johnlindquist/kit#1372] [Q&A] [answered]
How to perform some initialization operations when the main window opens

**Comment by johnlindquist** [comment]
@alkene0005 Can you describe what you're looking to do in a bit more detail? Run a script the same time you press the main shortcut? Modify the main window somehow?

**Reply by alkene5** [reply]
Set the default input method. I think the input method is in English when the window starts

**Reply by johnlindquist** [reply]
Understood. I can add a way to trigger a script when the main menu opens.

For example, running the main script:
 `~/.kit/main/index.js`
would trigger the following script "on init":
`~/.kenv/main/index.js`

Would that work for you?

**Reply by alkene5** [reply]
of course 😄

**Reply by johnlindquist** [reply, marked as answer]
@alkene0005 Can you try this build:

https://github.com/johnlindquist/kitapp/releases/tag/v1.99.76

Then in your ~/.kenv/.env, KIT_MAIN_HOOK pointing to the script you want to run when the main keyboard shortcut is pressed like this:
```
KIT_MAIN_HOOK_PATH=/Users/johnlindquist/.kenv/scripts/hook.js
```

You can make that path point to any script, but it will behave like a "terminal" script, so it won't be able to prompt for UI.

**Reply by alkene5** [reply]
yes, it works!! 😊 No problem at all, Looking forward to the 2.0 release
